File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceInitializeReply.java
package com.lumiyaviewer.lumiya.voice.common.messages;

import android.os.Bundle;
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage;
import javax.annotation.Nullable;

public class VoiceInitializeReply implements VoicePluginMessage {
    public final boolean appVersionOk;

    @Nullable
    public final String errorMessage;
    public final int pluginVersionCode;

    public VoiceInitializeReply(int pluginVersionCode, @Nullable String errorMessage, boolean appVersionOk) {
        this.pluginVersionCode = pluginVersionCode;
        this.errorMessage = errorMessage;
        this.appVersionOk = appVersionOk;
    }

    public VoiceInitializeReply(Bundle bundle) {
        this.pluginVersionCode = bundle.getInt("pluginVersionCode");
        this.errorMessage = bundle.getString("errorMessage");
        this.appVersionOk = bundle.getBoolean("appVersionOk");
    }

    @Override
    public Bundle toBundle() {
        Bundle bundle = new Bundle();
        bundle.putInt("pluginVersionCode", this.pluginVersionCode);
        bundle.putString("errorMessage", this.errorMessage);
        bundle.putBoolean("appVersionOk", this.appVersionOk);
        return bundle;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/voice/common/messages/VoiceInitializeReply.kt
package com.lumiyaviewer.lumiya.voice.common.messages

import android.os.Bundle
import com.lumiyaviewer.lumiya.voice.common.VoicePluginMessage

class VoiceInitializeReply : VoicePluginMessage {
    @JvmField
    val appVersionOk: Boolean
    @JvmField
    val errorMessage: String?
    @JvmField
    val pluginVersionCode: Int

    constructor(pluginVersionCode: Int, errorMessage: String?, appVersionOk: Boolean) {
        this.pluginVersionCode = pluginVersionCode
        this.errorMessage = errorMessage
        this.appVersionOk = appVersionOk
    }

    constructor(bundle: Bundle) {
        this.pluginVersionCode = bundle.getInt("pluginVersionCode")
        this.errorMessage = bundle.getString("errorMessage")
        this.appVersionOk = bundle.getBoolean("appVersionOk")
    }

    override fun toBundle(): Bundle = Bundle().apply {
        putInt("pluginVersionCode", pluginVersionCode)
        putString("errorMessage", errorMessage)
        putBoolean("appVersionOk", appVersionOk)
    }
}
